service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerMidEventTest.java

```diff
@@ -287,23 +298,98 @@ void shouldNotReturnAnErrorWhenAllChildrenHaveAFinalOrderIssuedButIssuingANonFin
                 createChild("Barry", true));
 
             AboutToStartOrSubmitCallbackResponse response = postMidEvent(
-                buildCaseDetails(null, chuckleBrothers, BLANK_ORDER, null), "add-final-order-flags");
+                buildCaseDetails(null, chuckleBrothers, BLANK_ORDER, null), callbackType);
 
             assertThat(response.getErrors()).isNull();
         }
     }
 
+    @Nested
+    class DischargeCaseOrderPreparation {
+
+        private final String callbackType = "prepare-selected-order";
+
+        @Test
+        void shouldReturnErrorWhenCurrentOrderIsDischargeOfCareOrderAndNoExistingCareOrders() {
+            GeneratedOrder order = order("Supervision order", "1 May 2020");
```

Comment:
Isn't there way to use keys in the checks? We define order types as:
CARE_ORDER

Reply:
done